application/cms/views.py

```diff
@@ -1226,7 +1191,10 @@ def edit_data_source(topic_slug, subtopic_slug, measure_slug, version, data_sour
         topic_slug, subtopic_slug, measure_slug, version
     )
 
-    data_source = DataSource.query.get(data_source_id)
+    try:
+        data_source = DataSource.query.get(data_source_id)
+    except DataError:  # non-integer data_source_id
```

Comment:
Can we avoid this by adding a 'converter' to the route, eg:

```python
@cms_blueprint.route(
    "/<topic_slug>/<subtopic_slug>/<measure_slug>/<version>/edit/data-sources/<int:data_source_id>", methods=["GET"]
)
```

See http://exploreflask.com/en/latest/views.html

Reply:
Yes, we could, although we don't really do that anywhere else, so it would be a little weird to just do it in that one place. Maybe.

Reply:
Feels like it'd be simpler than catching the exception? But we could save this for a future refactor?